iroh-net/src/discovery/local_swarm_discovery.rs

```diff
@@ -297,6 +368,13 @@ mod tests {
         use testresult::TestResult;
 
         #[tokio::test]
+        async fn local_swarm_discovery_smoke() -> TestResult {
+            // need to ensure that these tests run one after the other, otherwise
+            // they interfere with each other
+            test_local_swarm_discovery().await?;
+            test_subscribe().await
```

Comment:
ahhh, okay well, tbh these tests should not longer conflict at all regardless, due to the changes in `subscribe` since I initially wrote the tests.

passes fine locally to have these separated, so lets see if CI is mad about it for some reason.